lib/MusicBrainz/Server/Edit/Role/EditArtistCredit.pm

```diff
@@ -22,7 +26,21 @@ around initialize => sub {
         my $new_artist_credit = clean_submitted_artist_credits($opts{artist_credit});
 
         $opts{artist_credit} = $new_artist_credit;
-        delete $opts{artist_credit} if Compare($old_artist_credit, $new_artist_credit);
+        if (Compare($old_artist_credit, $new_artist_credit)) {
+            delete $opts{artist_credit}
+        } else {
+            my $new_artist_credit_full_name = '';
+            for my $part (@{ $new_artist_credit->{names} }) {
+              $new_artist_credit_full_name .= $part->{name} . $part->{join_phrase};
+            };
+
+            if (defined($new_artist_credit_full_name) && !(
+                has_oneline_string_length($new_artist_credit_full_name) &&
+                is_database_indexable_string($new_artist_credit_full_name)
+            )) {
```

Comment:
We do this check for three conditions often enough that it feels it should be its own function.

Reply:
Would a function for all of these conditions in our `Validation` module do?

Reply:
Sure :)

Reply:
I added `is_overlong_string` in https://github.com/metabrainz/musicbrainz-server/pull/3251/commits/abe06ef84d8655caffbdff15da2924593bbcb412.